src/main/java/eu/deltasource/internship/abankingsystem/model/Transaction.java

```diff
@@ -0,0 +1,176 @@
+package eu.deltasource.internship.abankingsystem.model;
+
+import eu.deltasource.internship.abankingsystem.enums.Currency;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+public class Transaction {
+
+    private int id;
+
+    private final BankAccount sourceAccount;
+
+    private final BankAccount targetAccount;
+
+    private final BankInstitution sourceBank;
+
+    private final BankInstitution targetBank;
+
+    private final double amount;
+
+    private final Currency sourceCurrency;
+
+    private final Currency targetCurrency;
+
+    private String transactionType;
+
+    private final LocalDate timestamp;
```

Comment:
Would be really good information to have also the time of the transaction

Reply:
Will be done ASAP.